cmd/util/trace.go

```diff
@@ -0,0 +1,58 @@
+package util
+
+import (
+	"context"
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+	"go.opentelemetry.io/otel/trace"
+	"google.golang.org/grpc"
+)
+
+func InitTracer(serviceName, otlpAddress string) (trace.TracerProvider, func(), error) {
+	ctx := context.Background()
+
+	res, err := resource.New(ctx,
+		resource.WithAttributes(
+			semconv.ServiceNameKey.String(serviceName),
+		),
+	)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to create resource: %w", err)
+	}
+
+	// Set up a trace exporter
+	exporter, err := otlptracegrpc.New(ctx,
+		otlptracegrpc.WithInsecure(),
+		otlptracegrpc.WithEndpoint(otlpAddress),
+		otlptracegrpc.WithDialOption(grpc.WithBlock()),
```

Comment:
Most of the time when I've seen grpc.WithBlock() mentioned in otel docs, it was for debugging. This could impact startup speed of the application as I believe it makes the connection to the otel endpoint block here.

Reply:
I agree that this shouldn't be a blocking call. Ideally we shouldn't have any impact on the overall ArgoCD performance by enabling traces.

Reply:
Removed this option so I will mark it as resolved.